src/wallet/wallet.cpp

```diff
@@ -1342,8 +1342,15 @@ bool CWalletTx::RelayWalletTransaction(CConnman* connman)
     {
         if (GetDepthInMainChain() == 0 && !isAbandoned() && InMempool()) {
             LogPrintf("Relaying wtx %s\n", GetHash().ToString());
-            RelayTransaction((CTransaction)*this);
-            return true;
+            if (connman) {
+                CInv inv(MSG_TX, GetHash());
+                connman->ForEachNode([&inv](CNode* pnode)
```

Comment:
Yes, but it's simple enough that I didn't bother factoring it out. No
reason not to, except that there's not an obvious place to stick it.
On Jun 14, 2016 19:26, "Jeremy Rubin" <notifications@github.com> wrote:

> In src/wallet/wallet.cpp
> <https://github.com/bitcoin/bitcoin/pull/8085#discussion_r67076355>:
>
> > @@ -1342,8 +1342,15 @@ bool CWalletTx::RelayWalletTransaction(CConnman* connman)
> >      {
> >          if (GetDepthInMainChain() == 0 && !isAbandoned() && InMempool()) {
> >              LogPrintf("Relaying wtx %s\n", GetHash().ToString());
> > -            RelayTransaction((CTransaction)*this);
> > -            return true;
> > +            if (connman) {
> > +                CInv inv(MSG_TX, GetHash());
> > +                connman->ForEachNode([&inv](CNode* pnode)
>
> Nit: appears correct, but isn't this the same code as in RelayTransaction?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin/bitcoin/pull/8085/files/a57b3f21e3228f838382e7510dc4e8f2bd023237..a01fa6c5a2e5884af64c0153dbfbd6a848128129#r67076355>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAZdE-sEqwxHrvI3wdUKtoHlUCgD1RM2ks5qLzi2gaJpZM4Ij8mn>
> .
>